UI/EncyclopediaDetailPanel.cpp

```diff
@@ -405,15 +447,59 @@ namespace {
             //for (auto str : GetStringTable().
 
         } else {
-            // list categories
-            std::map<std::string, std::vector<EncyclopediaArticle> >::const_iterator category_it =
-                encyclopedia.articles.find(dir_name);
-            if (category_it != encyclopedia.articles.end()) {
-                for (const EncyclopediaArticle& article : category_it->second) {
-                    sorted_entries_list.insert({UserString(article.name),
-                                                {LinkTaggedText(TextLinker::ENCYCLOPEDIA_TAG, article.name) + "\n",
-                                                 article.name}});
-                }
+            // Any content definitions (FOCS files) that define a pedia category
+            // should have their pedia article added to this category.
+            std::map<std::string, std::pair<std::string, std::string>> dir_entries;
+
+            // part types
+            for (const std::map<std::string, PartType*>::value_type& entry : GetPartTypeManager())
+                if (DetermineCustomCategory(entry.second->Tags()) == dir_name)
+                    dir_entries[UserString(entry.first)] = {VarText::SHIP_PART_TAG, entry.first};
```

Comment:
don't know why, but I get compile errors on all these lines. switching {} to std::make_pair() works though.